v1/operator.go

```diff
@@ -0,0 +1,139 @@
+	request := o.OperatorApi().DeleteOperatorRaftPeer(o.client.Ctx)
+
+	err := o.client.ExecNoResponseRequest(ctx, request)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *Operator) Autopilot(ctx context.Context) (*client.AutopilotConfiguration, error) {
+	request := o.OperatorApi().GetOperatorAutopilotConfiguration(o.client.Ctx)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(client.AutopilotConfiguration)
+	return &final, nil
+}
+
+func (o *Operator) UpdateAutopilot(
+	ctx context.Context,
+	cleanupdeadservers bool,
+	lastcontactthreshold string,
+	maxtrailinglogs int32,
+	serverstabilizationtime string,
+	enableredundancyzones bool,
+	disableupgrademigration bool,
+	enablecustomupgrades bool,
+) (*bool, error) {
+	request := o.OperatorApi().PutOperatorAutopilotConfiguration(o.client.Ctx)
+
+	// QUESTION: is there a better way to handle this?
+	// https://www.nomadproject.io/api-docs/operator/autopilot#update-autopilot-configuration
+	updateRequest := client.NewAutopilotConfiguration()
+	updateRequest.SetCleanupDeadServers(cleanupdeadservers)
+	updateRequest.SetLastContactThreshold(lastcontactthreshold)
+	updateRequest.SetMaxTrailingLogs(maxtrailinglogs)
+	updateRequest.SetServerStabilizationTime(serverstabilizationtime)
+	updateRequest.SetEnableRedundancyZones(enableredundancyzones)
+	updateRequest.SetDisableUpgradeMigration(disableupgrademigration)
+	updateRequest.SetEnableCustomUpgrades(enablecustomupgrades)
+
+	request = request.AutopilotConfiguration(*updateRequest)
+
+	//request.Execute() // returns *http.Response, error, but where's the bool!?!?
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(bool)
+	return &final, nil
+}
+
+func (o *Operator) AutopilotHealth(ctx context.Context) (*client.OperatorHealthReply, error) {
+	request := o.OperatorApi().GetOperatorAutopilotHealth(o.client.Ctx)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(client.OperatorHealthReply)
+	return &final, nil
+}
+
+func (o *Operator) Scheduler(ctx context.Context) (*client.SchedulerConfigurationResponse, *QueryMeta, error) {
+	request := o.OperatorApi().GetOperatorSchedulerConfiguration(o.client.Ctx)
+
+	result, meta, err := o.client.ExecQuery(ctx, request)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	final := result.(client.SchedulerConfigurationResponse)
+	return &final, meta, nil
+}
+
+// NOTE: there's no RejectJobRegistration?
+func (o *Operator) UpdateScheduler(
+	ctx context.Context,
+	scheduleralgorithm string,
+	memoryoversubscriptionenabled bool,
+	preemptionconfig *client.PreemptionConfig,
+) (*client.SchedulerSetConfigurationResponse, *WriteMeta, error) {
+	request := o.OperatorApi().PostOperatorSchedulerConfiguration(o.client.Ctx)
+
+	// QUESTION: how do I pass in the premptionconfig?
```

Comment:
Is this question still open or just stale? I think I see you setting it below.

Reply:
Just stale; I need to clean up my in-code questions/notes